src/proto/Aptos.proto

```diff
@@ -166,6 +166,8 @@ message SigningInput {
     LiquidStaking liquid_staking_message = 14;
     TokenTransferCoinsMessage token_transfer_coins = 15;
   }
+
+  optional string abi = 16;
```

Comment:
You can keep 5 indexes for the `transaction_payload` new variants that can come in the future.
So `abi` can be 21st field